### PR TITLE
feat(prompt): add filtering, sorting and pagination

### PR DESCRIPTION
**Pull request by RichardKruemmel**

### CODE REVIEW

**Comment by RichardKruemmel** [comment]
Test comment

File: web/src/features/prompts/server/prompt-router.ts [hunk 1]
```diff
@@ -19,40 +33,57 @@ export const CreatePrompt = z.object({
...
-        )
-        AND "project_id" = ${input.projectId}
-        ORDER BY name ASC`;
+      const orderByCondition =
+        input.orderBy?.column === "numberOfObservations"
+          ? Prisma.empty
+          : orderByToPrismaSql(input.orderBy, promptsTableCols);
+
+      const prompts = await ctx.prisma.$queryRaw<Array<Prompt>>(
+        generatePromptQuery(
```
Comment (marcklingen) [inline web/src/features/prompts/server/prompt-router.ts:69]: inline this if only used here?

File: web/src/features/prompts/server/prompt-router.ts [hunk 2]
```diff
@@ -497,3 +543,69 @@ export const createPrompt = async ({
...
+          GROUP BY name
+        )
+    AND "project_id" = ${projectId}
+  ${filterCondition}
+  ${orderCondition}
+  LIMIT ${limit} OFFSET ${page * limit};
+`;
+};
+
+function filterPromptsByCount(
```
Comment (marcklingen) [inline web/src/features/prompts/server/prompt-router.ts:572]: either do this via sql or not at all, this does not work anymore when we add pagination to this table

File: web/src/features/prompts/server/prompt-router.ts [hunk 3]
```diff
@@ -497,3 +543,69 @@ export const createPrompt = async ({
...
+        return p.observationCount <= value;
+      case ">=":
+        return p.observationCount >= value;
+      default:
+        return true;
+    }
+  });
+}
+
+function sortPromptsByObservationCount(
```
Comment (marcklingen) [inline web/src/features/prompts/server/prompt-router.ts:595]: api-level sorting breaks when we add pagination

Comment (marcklingen) [reply web/src/features/prompts/server/prompt-router.ts:595]: i'd reuse `orderByToPrismaSql`

File: web/src/features/prompts/server/prompt-router.ts (hunk 2 above)
Comment (marcklingen) [reply web/src/features/prompts/server/prompt-router.ts:572]: let's remove the filter and order of count alltogether

**Pull request merged**